### PR TITLE
Fix Mate - Maintenance Service App - V-0

### PR DESCRIPTION
**Pull request by anita00001**
# ⚙🛠FixMate - Maintenance Service App ⚒🔨

## About Project 📖
**Fix Mate** is committed to making everyday life more convenient by connecting users with trusted service providers. It is a one-stop platform for finding, booking, and reviewing services, all from the comfort of your home. Users can communicate with service providers through the app, discussing specific requirements, asking questions, and confirming details.

## Frontend PR:
[Frontend PR](https://github.com/anita00001/fix-mate-frontend/pull/36)

## Key Features 🔑
- **Show all Experts**
- **Reserve an Expert**
- **Show my Reservation**
- **Add an Expert**
- **Delete an Expert**
- **Add a Specialization**

## Version Requirements 🔺
- **ruby** >= 3.2.0
- **rails** >= 7.0
- **postgres** >- 15.3

## Back End
<details>
  <summary>Server</summary>
  <ul>
    <li><a href="https://www.ruby-lang.org/en/">Ruby</a></li>
    <li><a href="https://rubyonrails.org/">Ruby On Rails</a></li>
  </ul>
</details>

### Changes Made ♻
- [x] Setup API only Rails project
- [x] Setup Rubocop Checker
- [x] Draw ER Diagram
- [x] Create Migration and Schema following the ER diagram
- [x] Create a Model and Controller following the ER diagram
- [x] Create actions in the controller
- [x] Used `devise` and `devise-jwt` gems for authentication (Sign Up, Login, Logout)
- [x] Create API endpoints for experts, reservations, and specializations
- [x] Create an API documentation

## Useful links 🔗
✔ [Front-end Website](https://fixmate.netlify.app/)
✔ [Front-end Repository](https://github.com/anita00001/fix-mate-frontend)
✔ [Kanban Board Link](https://github.com/users/anita00001/projects/5)
✔ [Base Url](https://fixmate.onrender.com/)
✔ API endpoints
   - api/v1/experts
   - api/v1/reservations
   - api/v1/specializations
✔ [API documentation](https://fixmate.onrender.com/api-docs/index.html)
## Future Enhancements
- [ ] **Delete Specializations**
- [ ] **Add Admin Role**

## Assets
- ER-Diagram
- Kanban Board Screenshot
- App Authentication
- App Home Page

#

## Best Regards,
@anita00001  @Mofuhidy  @Theodoraldo  @AnsarIbrahim

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,291 +1,218 @@
...
--->
-
 <div align="center">
-  <!-- You are encouraged to replace this logo with your own! Otherwise you can also remove it. -->
-  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
-  <br/>
-
-  <h3><b>Microverse README Template</b></h3>
-
+  <h1>FixMate</h1>
```
Comment (Meltrust) [inline README.md:4]: - [x] You have included tests.   Kindly, add a "tests" section with specific instructions about how to correctly run your tests.  👍

Comment (Meltrust) [inline README.md:4]: - [x]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @anita00001,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  The API is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md (hunk 1 above)
Comment (AnsarIbrahim) [reply README.md:4]: Solve it

Comment (AnsarIbrahim) [reply README.md:4]: Solve It

**Review by Meltrust — APPROVED** [review]
#### Hi @anita00001,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Readme has been improved✔️ 
- Superfluous "test" folder has been removed✔️ 
- Great job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**